pkg/vault/test/foundry/BufferVaultPrimitive.t.sol

```diff
@@ -148,14 +148,77 @@ contract BufferVaultPrimitiveTest is BaseVaultTest {
 
         (, , IERC20[] memory tokens) = _getTokenArrayAndIndexesOfWaDaiBuffer();
         BaseVaultTest.Balances memory balancesBefore = getBalances(lp, tokens);
+        BufferBalance memory bufferBalanceBefore = _getBufferBalance();
 
         vm.prank(lp);
         (uint256[] memory pathAmountsOut, , ) = batchRouter.swapExactIn(paths, MAX_UINT256, false, bytes(""));
 
-        _checkWrapResults(balancesBefore, _wrapAmount, pathAmountsOut[0], false);
+        _checkWrapResults(balancesBefore, _wrapAmount, pathAmountsOut[0], bufferBalanceBefore, false);
     }
 
-    function testDepositWithBufferLiquidity() public {
+    function testDepositBufferMoreWrappedNotEnoughLiquidity() public {
+        // Initializes the buffer with an amount that's not enough to fulfill the deposit operation, so the vault has
+        // to interact with the ERC4626 protocol.
+
+        uint256 totalUnderlyingInBuffer = _wrapAmount / 2;
+        vm.startPrank(lp);
+        router.initializeBuffer(
+            IERC4626(address(waDAI)),
+            totalUnderlyingInBuffer / 3,
+            waDAI.previewDeposit((2 * totalUnderlyingInBuffer) / 3)
+        );
+        vm.stopPrank();
+
+        IBatchRouter.SwapPathExactAmountIn[] memory paths = _exactInWrapUnwrapPath(
```

Comment:
Late comment about this one: we're using the same method to wrap and unwrap, depending on token from and token to.
While this seems to be correct, you need to remember the order of the arguments to make it right. Here we're testing deposits for example, which means we are going to wrap (tokenFrom is underlying, tokenTo is wrapped).

I'd suggest splitting the method, having one for wrap and another one for unwrap, and you only specify the wrapper contract. That way it's harder to misuse the test helper, and it's immediately clear to the reader whether we're creating a path to wrap or unwrap in order to compare it to the test we're in.